logstreams/src/test/java/io/camunda/zeebe/logstreams/log/SequencerTest.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.logstreams.log;
+
+import io.camunda.zeebe.logstreams.impl.log.Sequencer;
+import io.camunda.zeebe.scheduler.ActorCondition;
+import io.camunda.zeebe.util.buffer.BufferWriter;
+import java.time.Duration;
+import java.util.List;
+import org.agrona.MutableDirectBuffer;
+import org.assertj.core.api.Assertions;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.mockito.Mockito;
+
+@SuppressWarnings("resource")
+@Execution(ExecutionMode.CONCURRENT)
+final class SequencerTest {
```

Comment:
:x: Please add tests to verify the sequencing of the batches is correct.

Reply:
I added three tests here: [3dcab4a](https://github.com/camunda/zeebe/pull/11165/commits/3dcab4afeefc7d6b64dac4bde40212d42ac45d4d)

Let me know what you think